pytorch_lightning/profiler/pytorch.py

```diff
@@ -422,12 +420,12 @@ def stop(self, action_name: str) -> None:
         if not _KINETO_AVAILABLE or self._emit_nvtx:
             return
 
-        if action_name in self.step_action_names:
+        if self.profiler is not None and action_name in self.STEP_FUNCTIONS:
             if self._schedule is not None:
-                self._schedule._current_action = action_name
+                self._schedule.pre_step(action_name)
 
             def on_trace_ready(profiler):
-                filename = f"{action_name}_{self.local_rank}"
+                filename = f"{self._stage}_{action_name}_{self.local_rank}"
```

Comment:
you can remove this

```suggestion
```

and pass below:

```python
handler = tensorboard_trace_handler(self.dirpath, self._prepare_filename(extension=""))
```

Reply:
Done !